gtsam/slam/ProjectionFactor.h

```diff
@@ -22,17 +22,20 @@
 
 #include <gtsam/nonlinear/NonlinearFactor.h>
 #include <gtsam/geometry/PinholeCamera.h>
+#include <gtsam/geometry/Pose3.h>
+#include <gtsam/geometry/Point3.h>
 #include <gtsam/geometry/Cal3_S2.h>
 #include <boost/optional.hpp>
 
 namespace gtsam {
 
   /**
-   * Non-linear factor for a constraint derived from a 2D measurement. The calibration is known here.
-   * i.e. the main building block for visual SLAM.
+   * Non-linear factor for a constraint derived from a 2D measurement. 
+   * The calibration is known here.
+   * The main building block for visual SLAM.
    * @addtogroup SLAM
    */
-  template<class POSE, class LANDMARK, class CALIBRATION = Cal3_S2>
+  template<class POSE=Pose3, class LANDMARK=Point3, class CALIBRATION = Cal3_S2>
```

Comment:
spacing before and after " = " are different for POSE and LANDMARK, vs CALIBRATION